app/helpers/application_helper.rb

```diff
@@ -55,6 +55,10 @@ def enable_google_analytics(cookies)
       !cookies[CookiesController::NO_ANALYTICS_MODE]
   end
 
+  def using_non_primary_url?
+    ENV["PRIMARY_HOST"].blank? || ENV["PRIMARY_HOST"] == request.host
```

Comment:
This looks backwards. Reading your code here in plain English, it seems to be saying:

"We are using a non-primary URL if either there is no primary host defined, or if there is a primary host and we are currently on it"

I think it should say:

"We are using a non-primary URL if there _is_ a primary host defined and we are _not_ on it."

Reply:
I think I've updated that now - I was getting confused when I lifted from the ApplicationController .

Reply:
It's still not quite there - because we only want to show the banner if PRIMARY_HOST _is_ defined. At the moment, if it's not defined, this check is effectively `nil != request.host`. Since request.host is always not nil, this will mean that if PRIMARY_HOST is not defined, this method will always return true.

Reply:
like this?

```
  def using_non_primary_url?
    ENV["PRIMARY_HOST"].nil? && ENV["PRIMARY_HOST"] != request.host
  end
```